sdk/inc/azure/iot/az_iot_hub_client.h

```diff
@@ -810,7 +808,7 @@ AZ_NODISCARD az_result az_iot_hub_client_properties_document_get_publish_topic(
  * @return An #az_result value indicating the result of the operation.
  * @retval #AZ_OK The topic was retrieved successfully.
  */
-AZ_NODISCARD az_result az_iot_hub_client_properties_update_get_publish_topic(
+AZ_NODISCARD az_result az_iot_hub_client_properties_get_reported_publish_topic(
```

Comment:
I'm including my raw comments from API review, where 100% agree to change refers to original `az_iot_hub_client_properties_update_get_publish_topic`.  The problem with making the [????] I reference below to be 'response` is that this is the same API used by like sending DeviceSerialNumber to Azure, and this isn't really responding to a cloud request.

I'm open for ???? feedback.  There's an impedance mismatch between MQTT, DTDLv2 nomenclature, and embedded SDK naming patterns that's making this surprisingly challenging.

---
100% agree this need to change. Per conversation and also starting at some, first the "get" totally needs to come 1st (after "namespace" bit) as verb. And then "publish_topic" needs to come last since that's conventction.

So we're agreed at least that it's az_iot_hub_client_properties_get_[????]_publish_topic. What's ????, though?

Possibilities: az_iot_hub_client_properties_get_update_publish_topic (or az_iot_hub_client_properties_get_updated_publish_topic)?

az_iot_hub_client_properties_get_reported_publish_topic // this is OK but kinda close to twin "reported" concept. az_iot_hub_client_properties_get_patch_publish_topic // issue is "patch" isn't really PnP nomenclature az_iot_hub_client_properties_get_send_publish_topic // "get_send" looks weird next to each other.

I'm going to try "az_iot_hub_client_properties_get_reported_publish_topic" since device is reporting up properties to cloud. We'll see how it looks once it hits broader review.

Reply:
ehhh yea i think it's fine. It gets the point across even if i blends concepts. I'd say the best option of the trio.